public/js/controllers/coding_progress_table_controller.js

```diff
@@ -88,6 +104,61 @@ export class CodingProgressTableController {
                     parentNode.attr("class", "coding-complete");
                 }
             });
+
+            // Attach event listeners to columns' checkbox & ability to filter table
+            d3.selectAll("input[type=checkbox]").each(function(d, i, n) {
+                d3.select(this).on("change", function() {
+                    if (d3.select(this).property("checked")) {
+                        CodingProgressTableController.selectedColumns = CodingProgressTableController.selectedColumns.filter(e => e !== this.nextElementSibling.innerText.trim());
+                        transform(column, sortInfo.order)
+                    } else {
+                        CodingProgressTableController.selectedColumns.push(this.nextElementSibling.innerText.trim())
+                        transform(column, sortInfo.order)
+                    }
+                })
+            });
+
+            // Enable the ability to filter table by selected keyword in dropdown menu 
+            d3.select("#keyword").on("change", function() {
+                CodingProgressTableController.keyword = this.options[this.selectedIndex].innerText.trim(); 
+                transform(column, sortInfo.order);
+            });
+
+             // Enable the ability to filter table by user's keyword input
+            let searchInputNode = document.getElementById("input-keyword");
+            d3.select("button#search").on("click", () => {
+                let value = searchInputNode.value.trim();
+                if (value) {
+                    CodingProgressTableController.keyword = value; 
+                    transform(column, sortInfo.order);
+                } else {
+                    alert("Enter keyword...")
+                }
+            })
+            searchInputNode.addEventListener("keydown", function onEvent(event) {
+                if (event.key === "Enter") {
+                    let value = searchInputNode.value.trim();
+                    if (value) {
+                        CodingProgressTableController.keyword = value; 
+                        transform(column, sortInfo.order);
+                    } else {
+                        alert("Enter keyword...")
+                    }
+                }
+            });
+
+            // View all datasets on clicking `view all` button
```

Comment:
View all button should also reset the drop-down back to **Filter By Active Projects**

Reply:
Updated thanks